micro-rdk-nmea-macros/src/composition.rs

```diff
@@ -138,17 +129,22 @@ impl PgnComposition {
         let error_ident = quote! {#crate_ident::parse_helpers::errors::NmeaParseError};
         let mrdk_crate = crate::utils::get_micro_rdk_crate_ident();
         quote! {
+
             impl #impl_generics #name #src_generics #src_where_clause {
-                pub fn from_cursor(mut cursor: #crate_ident::parse_helpers::parsers::DataCursor, source_id: u8) -> Result<Self, #error_ident> {
+                #(#attribute_getters)*
```

Comment:
This block is now *so close* to what is in `into_fieldset_token_stream`. Is there a way to unify it?

Reply:
The common statements in the meat of the functions are already pretty unified, I'm not sure what further abstraction is valuable?

Reply:
Nevermind, I took a closer look and there is more variation than I'd realized between the two.